frappe/integrations/doctype/social_login_key/social_login_key.py

```diff
@@ -80,6 +88,8 @@ def validate(self):
 			frappe.throw(
 				_("Please enter Client Secret before social login is enabled"), exc=ClientSecretNotSetError
 			)
+		if self.social_login_provider == "Keycloak":
```

Comment:
Seems fine, but this block shouldn't be needed as per my understanding. Setting the path in the provider configuration should be enough

Reply:
No, we need that. Not everyone will have the same URL as their base URL and the endpoint. And also API endpoint requires base URL in it, because API endpoint is not editable , we are making it (editable) according to the base URL.

Reply:
The library automatically adds the base URL, that's why it shouldn't be required. That's how it works for other providers like Frappe.

Reply:
@akhilnarang I'm sorry to say this, but the library does NOT add the base URL. If you would test this yourself you would be able to confirm this. It is well documented and many others have faced this EXACT same issue. The base URL MUST be added in order for it to work with Keycloak. Either frappe needs to fix this fundamental issue or else this code is required. 

If you do not believe me, please verify it yourself. You can also see how other people need to manually add the base url in the following issue.

https://github.com/frappe/erpnext/issues/35542#issuecomment-1657126920

Reply:
Read up some more: so basically this is because the realm is dynamic?

Could you share some complete URLs for the API endpoint, authorize URL, access token URL

Reply:
And for this your base URL is just `http://localhost:8080`?

Reply:
http://localhost:8080/realms/master

Reply:
Thanks for the information, will experiment a bit.

Reply:
Sure but can you do it quick , we really need this to be done fast.

Reply:
@akhilnarang  Can you please review and accept it.

Reply:
Haven't gotten time to check some things that I wanted to yet.

Reply:
Ok, let me know if you need any help setting up keycloak to test it.

Reply:
@pa1narendra I've pushed a commit, can you test that, with the base URL set like this: `http://localhost:8080/realms/{realm_name}`
![image](https://github.com/user-attachments/assets/8c7d7db2-2506-45aa-96bc-d5fe12ee80a5)

Reply:
@akhilnarang  Is that working after you made those changes ???

Reply:
@pa1narendra it'll work now, do check.

I moved the URL setting to another place where it seemed more appropriate, and added a comment explaining why its needed.

Reply:
@akhilnarang  Can you test it once again ? I tried and it didn't worked for me. Or is there anything that I want to know while configure and running ?, I am getting the same Email not verified error and also the API Endpoint URL is not updated on save. Can you tell me what changes I have to make it work ?



![2024-09-19-170052_screenshot](https://github.com/user-attachments/assets/26082c10-f196-4a97-ab65-9775e2ce1e6c)

Reply:
![image](https://github.com/user-attachments/assets/6d0ebbf8-eef3-42ea-b68e-145cae0fa4d5)

This was my setup and it worked

Reply:
Is the API Endpoint getting updated on save ??

Reply:
My bad , it is working somehow, I actually forgot to check the email verified and it is perfectly working , I can be able to login. Thank you @akhilnarang .